src/server/service/user.ts

```diff
@@ -30,7 +30,51 @@ export default class UserService implements IUserService {
     return hash;
   }
 
-  async create(dto: CreateUserDTO): Promise<User> {
+  private checkRoles(role: string): void {
+    const ROLES = [
+      Role.Admin as string,
+      Role.Manager as string,
+      Role.User as string,
+    ];
+
+    if (!ROLES.includes(role)) {
+      throw new InvalidUserRole();
+    }
+  }
```

Comment:
I don't think this validation belongs to this layer of the application.

The service should assume that the data received have a correct shape, if something doesn't look as expected the receiver of such data (in this case the HTTP handler), should make sure data looks as expected.

For instance, if I send `{ "foo": "bar" }` to the HTTP handler to create a user, such object will get here, that shouldn't happen, we should be able to avoid arbitrary data to get here, that way our application is more robust and secure.